setpass/api.py

```diff
@@ -166,5 +168,46 @@ def add(user_id):
     return Response(response=user.token, status=200)
 
 
+@wsgi.app.route('/reset', methods=['GET'])
+def view_reset_form():
+    return render_template('reset_form.html')
+
+
+@wsgi.app.route('/reset', methods=['POST'])
+def reset_password():
```

Comment:
I just did some quick reading of the first Google results on this, and I think you would do a PUT on some user-specific endpoint.  So for the actual password reset.  But, in this case, you are creating a new password reset request.  The password is not actually reset in this part.

Perhaps it is just a clarity/naming problem, I could change the name of the endpoint and function to something like "request-reset" ?

I copy-pasted the set_password() function and poked at it until it worked.  I didn't have to change the POST part to make it work. :-)